crates/solver/src/settlement_simulation.rs

```diff
@@ -48,41 +54,47 @@ pub async fn simulate_and_estimate_gas_at_current_block(
         return Ok(Vec::new());
     }
 
-    let web3 = web3::Web3::new(web3::transports::Batch::new(web3.transport().clone()));
+    let web3 = web3::Web3::new(shared::transport::buffered::Buffered::new(
+        web3.transport().clone(),
+    ));
+    let contract_with_buffered_transport = GPv2Settlement::at(&web3, contract.address());
     let mut results = Vec::new();
     for chunk in settlements.chunks(SIMULATE_BATCH_SIZE) {
-        let calls = chunk
+        let txs = chunk
             .iter()
             .map(|(account, settlement)| {
-                let tx = settle_method(gas_price, contract, settlement.clone(), account.clone()).tx;
-                let resolved_gas_price = tx
-                    .gas_price
-                    .map(|gas_price| gas_price.resolve_for_transaction())
-                    .unwrap_or_default();
-                let call_request = CallRequest {
-                    from: tx.from.map(|account| account.address()),
-                    to: tx.to,
-                    gas: None,
-                    gas_price: resolved_gas_price.gas_price,
-                    value: tx.value,
-                    data: tx.data,
-                    transaction_type: resolved_gas_price.transaction_type,
-                    access_list: None,
-                    max_fee_per_gas: resolved_gas_price.max_fee_per_gas,
-                    max_priority_fee_per_gas: resolved_gas_price.max_priority_fee_per_gas,
+                settle_method(
+                    gas_price,
+                    &contract_with_buffered_transport,
+                    settlement.clone(),
+                    account.clone(),
+                )
+                .tx
+            })
+            .collect::<Vec<_>>();
+        let mut access_lists = access_list_estimator
+            .estimate_access_lists(txs.as_slice())
```

Comment:
I'm wondering if it would be cleaner to create the access lists when we also create the rest of the transaction itself and then pass it around to the simulation here and to the submitter later. That way the access list only has to be created once. Also it feels more proper to me that the access list would be part of the transaction and not be created on demand in several places.

Reply:
Access list is something that is theoretically dependent on the current network state, so it makes sense to me to be created at the moment of using. That is the reason why I estimate access list in every iteration of the submission loop, instead of doing it only once before entering the loop. Not sure about this though, if others think we should calculate it only once, I can change it

Reply:
Makes sense to me. It reduces the change of paying for storage access you don't end up using.

Reply:
Refactored these two functions to receive already prepared access list. Access lists are now created once in driver.